### PR TITLE
Implement Whatsapp with Vonage using Deno

### LINKED ISSUE #201: 🚀 Feature: Add WhatsApp with Vonage for Deno
Appwrite Functions allow you to extend and customize your Appwrite server functionality by executing your custom code. 🤩 You can learn more at our official [Appwrite Functions docs](https://appwrite.io/docs/functions).

Your task is to implement the WhatsApp with Vonage function in Deno. You can look at the existing [Appwrite Functions Templates](https://github.com/appwrite/templates) in another coding language.

Your function should behave and be implemented similarly to existing WhatsApp with Vonage Node.js template.

### Tasks summary:

- [ ] Fork & clone Appwrite's [templates](https://github.com/appwrite/templates) repository
- [ ] Create a branch `feat-implement-whatsapp-with-vonage-deno`
- [ ] Write code for the function in folder `deno/whatsapp-with-vonage/`. 
- [ ] Test the function using Appwrite, **provide a video**.
- [ ] Write a `README.md` explaining what the function does, how to use it, and customize it. Here is [an example](https://github.com/appwrite/templates/tree/main/node/whatsapp-with-vonage).
- [ ] Submit a pull request to our [templates](https://github.com/appwrite/templates) repository

If you need any help, contact us on our [Discord server](https://discord.gg/GSeTUeA).

Are you ready to work on this issue? 🤔 Let us know, and we will assign it to you 😊

**Happy Appwriting!**

### PR DESCRIPTION
**Pull request by ShamanthOP**
## What does this PR do?

Implement ```whatsapp-with-vonage``` template in Deno.

## Test

https://github.com/appwrite/templates/assets/69506046/8eed3095-17df-4951-88e3-4b3712f22205

image

## Related PRs and Issues

Closes #201 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: deno/whatsapp-with-vonage/src/main.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    req: any;
+    res: any;
+    log: (msg: any) => void;
+    error: (msg: any) => void;
+};
+
+export default async ({ req, res, log, error }: Context) => {
+    throwIfMissing(Deno.env.toObject(), [
+        "VONAGE_API_KEY",
+        "VONAGE_ACCOUNT_SECRET",
```
Comment (loks0n) [inline deno/whatsapp-with-vonage/src/main.ts:18]: Can we change this to VONAGE_API_SECRET throughout including readme?

File: deno/whatsapp-with-vonage/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        "VONAGE_SIGNATURE_SECRET",
```
Comment (loks0n) [inline deno/whatsapp-with-vonage/src/main.ts:19]: Can we change this to VONAGE_API_SIGNATURE_SECRET throughout including readme?

**Review by loks0n — CHANGES_REQUESTED** [review]

File: deno/whatsapp-with-vonage/src/main.ts (hunk 1 above)
Comment (ShamanthOP) [reply deno/whatsapp-with-vonage/src/main.ts:18]: Sure

File: deno/whatsapp-with-vonage/src/main.ts (hunk 2 above)
Comment (ShamanthOP) [reply deno/whatsapp-with-vonage/src/main.ts:19]: Yeah

File: deno/whatsapp-with-vonage/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import { crypto } from "https://deno.land/std@0.203.0/crypto/mod.ts";
+import { encodeHex } from "https://deno.land/std@0.203.0/encoding/hex.ts";
+
```
[the remark this answers is not in the document]
Comment (ShamanthOP) [reply deno/whatsapp-with-vonage/src/main.ts:3]: Sure

File: deno/whatsapp-with-vonage/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { jwtVerify } from "https://deno.land/x/jose@v4.15.2/index.ts";
+
```
[the remark this answers is not in the document]
Comment (ShamanthOP) [reply deno/whatsapp-with-vonage/src/main.ts:5]: Sure

**Comment by ShamanthOP** [comment]
Hey @loks0n I've added the requested changes.

File: deno/whatsapp-with-vonage/src/main.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        Deno.env.get("VONAGE_SIGNATURE_SECRET")
+    );
+    const { payload } = await jwtVerify(token, secret, {
+        algorithms: ["HS256"],
+    });
+
+    try {
+        throwIfMissing(payload, ["payload_hash"]);
+    } catch (err) {
+        if (err instanceof Error)
```
[the remark this answers is not in the document]
Comment (ShamanthOP) [reply deno/whatsapp-with-vonage/src/main.ts]: It types the `err.message` to be `string`, without which it'd be `any`. Shall I remove the if statement?

**Review by loks0n — APPROVED** [review]

**Comment by gewenyu99** [comment]
Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship. 

Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

**Comment by ShamanthOP** [comment]
It's `shamanth8360`

**Pull request merged**